macosx/BlocklistDownloaderViewController.mm

```diff
@@ -103,7 +103,7 @@ - (void)setFailed:(NSString*)error
 
     alert.informativeText = error;
 
-    [alert beginSheetModalForWindow:self.fPrefsController.window completionHandler:^(NSModalResponse returnCode) {
+    [alert beginSheetModalForWindow:self.fPrefsController.window completionHandler:^(__unused NSModalResponse returnCode) {
```

Comment:
Here and elsewhere: comment out instead, in line with the overall style:
```suggestion
    [alert beginSheetModalForWindow:self.fPrefsController.window completionHandler:^(NSModalResponse /*returnCode*/) {
```

Reply:
`__unused` is made for that, and unnamed parameters isn't compatible with pure Objective-C (I know we're using Objective-C++), which makes it less portable for no benefits.

Reply:
Any identifier starting with two underscores is reserved by the standard; since the standard doesn't document `__unused` and we're not defining it ourselves (because we can't, based on the standard), we should consider it an implementation detail and shouldn't be using it. It might be "made for that" by people writing the system headers, but it doesn't give you the right to use it in your code.

Since this macro suppresses the warning, it's less strict than commenting the name out. It no longer warns you if the variable is unused, but it doesn't warn you when the variable is in fact used (and so the suppression is meaningless) either. If you understand that the code is compiled as Obj-C++, why do you insist on applying an Obj-C solution to the problem?

FYI we used to have our own `TR_UNUSED` macro but it was removed as part of transition to C++ (see #1918 and #2033).

Reply:
It's in cdefs.h:
![Capture d’écran 2022-12-19 à 00 26 14](https://user-images.githubusercontent.com/839992/208308838-8853b8fa-6c32-44f6-9ea7-55d956a8c713.png)

It's basically the equivalent of `[[maybe_unused]]`, but for Objective-C.

And to get the warning for using a normally unused variable, we can add `deprecated`:
```
#define tr_unused __attribute__((deprecated, unused))
```

To my knowledge, Apple doesn't provide a single API in C++ (only C, ObjC or Swift). So Objective-C++ usage shall preferably be limited maybe to short functions that wraps C++ API to an Objective-C API.

Anyway, not a blocker, it all [only exists to silent that warning](https://github.com/isocpp/CppCoreGuidelines/blob/036324/CppCoreGuidelines.md#f9-unused-parameters-should-be-unnamed):
>F.9: Unused parameters should be unnamed
>**Reason**
>Readability. **Suppression of unused parameter warnings.**

So I've done the replacement as you wished.